opentelemetry-auto-instrumentation/lib/opentelemetry-auto-instrumentation.rb

```diff
@@ -0,0 +1,153 @@
+
+        OpenTelemetry::SDK.configure do |c|
+          c.resource = detect_resource_from_env
+          if required_instrumentation.empty?
+            c.use_all # enables all instrumentation!
+          else
+            required_instrumentation.each do |instrumentation|
+              c.use instrumentation
+            end
+          end
+        end
+        OpenTelemetry.logger.info { 'Auto-instrumentation initialized' }
+      rescue StandardError => e
+        warn "Auto-instrumentation failed to initialize. Error: #{e.message}"
+      end
+    end
+  end
+
+  # ref: https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/boot/strap.rb
+  def require(...)
+    super
+    OTelInitializer.require_otel
+  end
+end
+
+require 'bundler'
+
+aws = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s.include?('aws')
+container = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s.include?('container')
+google_cloud_platform = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s.include?('google_cloud_platform')
+azure = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s.include?('azure')
+
+# set OTEL_OPERATOR to false if not in autoinstrumentation-ruby image, default to /otel-auto-instrumentation-ruby
+# /otel-auto-instrumentation-ruby is set in opentelemetry-operator ruby.go
+operator_gem_path = ENV['OTEL_RUBY_OPERATOR'].nil? || ENV['OTEL_RUBY_OPERATOR'] == 'true' ? '/otel-auto-instrumentation-ruby' : nil
+additional_gem_path = operator_gem_path || ENV['OTEL_RUBY_ADDITIONAL_GEM_PATH'] || Gem.dir
+$stdout.puts "Loading the additional gem path from #{additional_gem_path}" if ENV['OTEL_RUBY_AUTO_INSTRUMENTATION_DEBUG'] == 'true'
+
+# use OTEL_RUBY_UNLOAD_LIBRARY to avoid certain gem preload (esp. google protobuf)
+# e.g export OTEL_RUBY_UNLOAD_LIBRARY=google-protobuf;googleapis-common-protos-types
+unload_libraries = ENV['OTEL_RUBY_UNLOAD_LIBRARY'].to_s.split(';')
+
+loaded_library_file_path = Dir.glob("#{additional_gem_path}/gems/*").select do |file_path|
+  gem_name = file_path.match(%r{/gems/([^/]+)-\d})[1]
+  (file_path.include?('opentelemetry') || file_path.include?('google')) && !unload_libraries.include?(gem_name)
+end
+
+$stdout.puts "Loaded Library File Paths #{loaded_library_file_path.join(',')}" if ENV['OTEL_RUBY_AUTO_INSTRUMENTATION_DEBUG'] == 'true'
+
+loaded_library_file_path.each do |file_path|
+  $LOAD_PATH.unshift("#{file_path}/lib")
+end
+
+$stdout.puts "$LOAD_PATH after unshift: #{$LOAD_PATH.join(',')}" if ENV['OTEL_RUBY_AUTO_INSTRUMENTATION_DEBUG'] == 'true'
+
+require 'opentelemetry-sdk'
+require 'opentelemetry-instrumentation-all'
+require 'opentelemetry-exporter-otlp'
+
+require 'opentelemetry-resource-detector-container' if container
+require 'opentelemetry-resource-detector-google_cloud_platform' if google_cloud_platform
+require 'opentelemetry-resource-detector-azure' if azure
+require 'opentelemetry-resource-detector-aws' if aws
+
+Bundler::Runtime.prepend(OTelBundlerPatch)
+
+Bundler.require if ENV['OTEL_RUBY_REQUIRE_BUNDLER'] == 'true'
```

Comment:
what is the relationshiop between using `Bundler.require` and if someone uses `bundler/setup`?

Will this work as expected?